SeQuant/core/eval/result.hpp

```diff
@@ -361,11 +367,22 @@ auto biorthogonal_nns_project_ta(TA::DistArray<Args...> const& arr,
 
   using numeric_type = typename TA::DistArray<Args...>::numeric_type;
 
-  size_t factorial_ket = 1;
-  for (size_t i = 2; i <= ket_rank; ++i) {
-    factorial_ket *= i;
+  std::vector<numeric_type> cleanup_weights;
+
+  if (ket_rank >= 3) {
+    if (ket_rank < 6) {
```

Comment:
raw numerals like `6` should ideally not be used directly. Instead, create a variable (can be constexpr) whose name describes what this is. E.g. `max_rank_hardcoded_nns_projector`